### PR TITLE
dont check uninitialzed vars, change info to error

### LINKED ISSUE #18: Bug in med_merge_mod
The variables ungriddedUbound_output, ungriddedUBound_input, gridtofieldmap_input and gridtofieldmap_output are not initialized unless lrank==2.

### PR DESCRIPTION
**Pull request by jedwards4b**
Change Description: Add a check lrank==2 before testing variables. 
Fixes #18 

Testing Completed: scripts_regression_tests.py
(Minimum testing: CIME_DRIVER=nuopc scripts_regression_tests.py)

CIME HASH: b51b397d

### CODE REVIEW

**Comment by jedwards4b** [comment]
All scripts_regression_tests.py pass except Z_FullSystemTest and T_TestRunRestart.  These failures are unrelated to this PR.

File: mediator/med_merge_mod.F90 [hunk 1]
```diff
@@ -290,16 +293,18 @@ subroutine med_merge_auto_field(merge_type, FBout, FBoutfld, FB, FBfld, FBw, fld
...
-       return
-    else if (gridToFieldMap_input(1) /= gridToFieldMap_output(1)) then
-       call ESMF_LOGWrite(trim(subname)//"gridToFieldMap_input not equal to gridToFieldMap_output", ESMF_LOGMSG_INFO)
-       rc = ESMF_FAILURE
-       return
-    end if
-
+    if (lrank == 2) then
+       if (ungriddedUBound_output(1) /= ungriddedUBound_input(1)) then
+          write(errmsg,*) trim(subname),"ungriddedUBound_input (",ungriddedUBound_input(1),") not equal to ungriddedUBound_output (",ungriddedUBound_output(1),")"
```
Comment (mvertens) [inline mediator/med_merge_mod.F90]: Can you please check that this line is not longer than 132 characters - and break it up if it is

File: mediator/med_merge_mod.F90 [hunk 2]
```diff
@@ -177,7 +178,8 @@ subroutine med_merge_auto(compout_name, FBOut, FBfrac, FBImp, fldListTo, FBMed1,
 
                          if (FB_FldChk(FBImp(compsrc), trim(merge_field), rc=rc)) then
                             call med_merge_auto_field(trim(merge_type), &
-                                 FBOut, fldname, FB=FBImp(compsrc), FBFld=merge_field, FBw=FBfrac, fldw=trim(merge_fracname), rc=rc)
+                                 FBOut, fldname, FB=FBImp(compsrc), FBFld=merge_field, &
```
Comment (mvertens) [inline mediator/med_merge_mod.F90]: Why is there a blank line here?

Comment (jedwards4b) [reply mediator/med_merge_mod.F90]: removed

File: mediator/med_merge_mod.F90 (hunk 1 above)
Comment (jedwards4b) [reply mediator/med_merge_mod.F90]: fixed

**Pull request merged**